### ISSUE TITLE
clhep@2.3.1.1 broken with gcc@4.8.5

### ISSUE THREAD

**Issue by briedel** [spack/spack#6756]
I cannot get `clhep@2.3.1.1` to compile with standard EL7 compiler (`gcc@4.8.5`) due to ` c++1y` extension, see 

```
1 error found in build log:
     [ ... ]
     13    -- Detecting CXX compiler ABI info
     14    -- Detecting CXX compiler ABI info - done
     15    -- Detecting CXX compile features
     16    -- Detecting CXX compile features - done
     17    -- libraries will be installed in /cvmfs/oasis.opensciencegrid.org/osg/spack/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/clhep-2.3.1.1-4jpsm36u3oo4s4nx5yxzfutgd3yrbw7t/lib
     18    -- cmake build type is RelWithDebInfo
  >> 19    CMake Error at cmake/Modules/ClhepVariables.cmake:149 (message):
     20      c++1y extension is not available for
     21      /cvmfs/oasis.opensciencegrid.org/osg/spack/lib/spack/env/gcc/g++
     22    Call Stack (most recent call first):
     23      cmake/Modules/ClhepVariables.cmake:196 (_clhep_verify_cxx1y)
     24      cmake/Modules/ClhepVariables.cmake:255 (_clhep_check_cxxstd)
     25      CMakeLists.txt:85 (clhep_set_compiler_flags)
```
This works with gcc@5.3.0 and gcc@4.9.4. clhep@2.3.1.1 is required for the "standard" geant4 install, e.g. `geant4@10.02.p02`. 

@gartung is there a workaround?

**Comment by gartung** [comment]
No. You have to use a compiler that support -std=c++1y. gcc 4.9.x+ and clang 8.1+ support this standard but not gcc4.8.5.

**Comment by briedel** [comment]
Thanks. I will add restrictions to the `clhep` package in that case

**Comment by briedel** [comment]
`clhep@2.3.1.1` and `clhep@2.3.4.4` at least compiles with `gcc@4.8.5`. Why does `geant4@10.02.p02` require `c++14` version?

```
[centos@cvmfs-rhel7-build-machine ~]$ spack spec geant4
Input spec
--------------------------------
geant4

Concretized
--------------------------------
geant4@10.02.p02%gcc@5.3.0 build_type=RelWithDebInfo +qt arch=linux-rhel7-x86_64
    ^clhep@2.3.1.1%gcc@5.3.0 build_type=RelWithDebInfo ~cxx11+cxx14 arch=linux-rhel7-x86_64
[...]
```

**Comment by gartung** [comment]
It uses the c++14 standard so CLHEP has to match it. Look in the cmake options. I think there is one to enable c++1y at least.

**Comment by briedel** [comment]
The website claims it supports `gcc@4.8.5`. Is the `C++14` support required or "nice to have"?

**Comment by gartung** [comment]
Something probably got lost in the transition to passing compiler flags on the command line. What it probably should say is

depends_on(clhep~cxx11+cxx14) if c++14 or c++1y in spec
depends_on(clhep+cxx11~cxx14) if c++11 in spec

In fact the clhep package should be changed as well to set cmake flags based on what is in the spec.

**Comment by briedel** [comment]
Okay, I will make the changes tomorrow and make a PR. Should I update the versions while I am at it?

**Comment by gartung** [comment]
Please do.

**Comment by gartung** [comment]
I think the c++14 std was picked because that is the standard CMS software is built with and the libraries have to match stds.

**Comment by briedel** [comment]
@gartung  is there a reason you are pulling in the data files explicitly [here](https://github.com/gartung/cmssw-spack/blob/master/packages/geant4/package.py#L55-L62)? Should I do the same?

**Comment by gartung** [comment]
That is up to you. I am copying the way the geant data is handled by CMS. This allows multiple patch versions of geant which all share the same data files.

**Comment by alalazo** [comment]
Should be closed by #6956
